File: Assets/Scripts/Entity/GameEntity.cs
using UnityEngine;
using System.Collections;

public abstract class GameEntity
{
    public GameObject gameObject;
    public Transform transform;


    private bool m_bDisposed = false;


    public GameEntity(string prefabName)
    {
        gameObject = MgrRes.newObject(prefabName) as GameObject;
        transform = gameObject.transform;
    }


    public virtual void dispose(){
        m_bDisposed = true;
        GameObject.Destroy(gameObject);
        gameObject = null;
        transform = null;
    }


    public bool isDisposed()
    {
        return m_bDisposed;
    }


    public virtual void setPosition(float x, float y)
    {
        setPosition(new Vector3(x, y, 0));
    }


    public virtual void setPosition(Vector3 p)
    {
        transform.position = p;
    }


    public Vector3 getPosition()
    {
        return transform.position;
    }


    public bool collisionWith(GameEntity entity)
    {
        //entity.gameObject.transform.
        // 客户端貌似不做判断
        return false;
    }


    public void show()
    {
        if(!m_bDisposed)
            gameObject.SetActive(true);
    }


    public void hide()
    {
        if(!m_bDisposed)
            gameObject.SetActive(false);
    }



    public void update()
    {
        if(!m_bDisposed)
            onUpdate();
    }



    public abstract void onUpdate();
}

File: ServerNetty/src/gege/game/GameEntity.java
package gege.game;

public abstract class GameEntity {
	public float x;
	public float y;
	private float w = 0.6f;
	private float h = 0.6f;
	
	// 用于碰撞
	private float halfw = w / 2;
	private float halfh = h / 2;
	
	
	protected int m_state;
	
	
	private boolean m_bDisposed = false;
	
	
	
	public void setSize(float w, float h){
		this.w = w;
		this.h = h;
		halfw = w / 2;
		halfh = h / 2;
	}
	
	
	
	protected void setPosition(float x, float y){
		this.x = x;
		this.y = y;
	}
	
	
	protected void setNextState(int state){
		m_state = state;
	}
	
	
	public boolean inState(int state){
		return m_state == state;
	}
	
	
	public boolean tryCollision(GameEntity entity){
		if(x - halfw > entity.x + entity.halfw)
			return false;
		if(x + halfw < entity.x - entity.halfw)
			return false;
		if(y - halfh > entity.y + entity.halfh)
			return false;
		if(y + halfh < entity.y - entity.halfh)
			return false;
		
		return true;
	}
	
	
	public void dispose(){
		m_bDisposed = true;
	}
	
	
	public boolean isDisposed(){
		return m_bDisposed;
	}
	
	
	final public void update(){
		if(m_bDisposed)
			return;
		
		onUpdate();
	}
	
	
	public abstract void onUpdate();
}
